google/cloud/storage/internal/sha256_hash.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/sha256_hash.h"
+#include <openssl/evp.h>
+#include <openssl/sha.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace internal {
+
+std::vector<std::uint8_t> Sha256Hash(std::string const& str) {
+  SHA256_CTX sha256;
+  SHA256_Init(&sha256);
+  SHA256_Update(&sha256, str.c_str(), str.size());
+
+  unsigned char hash[SHA256_DIGEST_LENGTH];
+  SHA256_Final(hash, &sha256);
+  return {hash, hash + sizeof(hash)};
+}
+
+std::string HexEncode(std::vector<std::uint8_t> const& str) {
+  std::string result;
+  for (auto c : str) {
+    char buf[16];
```

Comment:
Nit: It doesn't really matter of course, but 16 makes me wonder if something else is going on.

Reply:
Would you prefer another number?  I have no opinion as long as it is (a) large enough, (b) not too large... Your call.

Reply:
Sure ... 3, which I would probably write as `sizeof("ff")`.  But, just a nit ... your call.

Reply:
Done.